WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageParentViewModel.kt

```diff
@@ -128,4 +152,74 @@ class PageParentViewModel
         }
         return children + grandchildren
     }
+
+    fun onSearchExpanded(restorePreviousSearch: Boolean) {
+        if (isSearchExpanded.value != true) {
+            if (!restorePreviousSearch) {
+                clearSearch()
+            }
+            _isSearchExpanded.value = true
+        }
+    }
+
+    fun onSearchCollapsed() {
+        _isSearchExpanded.value = false
+        clearSearch()
+
+        launch {
+            delay(SEARCH_COLLAPSE_DELAY)
+        }
+    }
+
+    private fun clearSearch() {
+        _lastSearchQuery = ""
+        _searchPages.postValue(null)
+        _isSaveButtonVisible.postOnUi(false)
+        _currentParent.isSelected = false
+    }
+
+    fun onSearch(searchQuery: String, delay: Long = SEARCH_DELAY) {
+        searchJob?.cancel()
+        if (searchQuery.isNotEmpty()) {
+            searchJob = launch {
+                delay(delay)
+                searchJob = null
+                if (isActive) {
+                    _lastSearchQuery = searchQuery
+                    val result = search(searchQuery)
+                    _searchPages.postValue(result)
+                }
+            }
+        } else {
+            clearSearch()
+        }
+    }
+
+    private suspend fun search(
+        searchQuery: String
+    ): List<PageItem> = withContext(defaultDispatcher) {
+        val tempPages = _pages.value
+        if (tempPages != null) {
```

Comment:
NitPick: We might want to consider using `let` here.

```
_pages.value?.let {
...
}
```
Wdyt?

Reply:
NitPick accepted :) Changed & pushed